core/src/main/java/feast/core/config/WebSecurityConfig.java

```diff
@@ -40,7 +40,7 @@ protected void configure(HttpSecurity http) throws Exception {
 
     // Bypasses security/authentication for the following paths
     http.authorizeRequests()
-        .antMatchers("/actuator/**", "/metrics/**")
+        .antMatchers("/actuator/**", "/metrics/**", "/api/v1/**")
```

Comment:
I think this needs to be dealt with.  It's a lot of work to add authentication and prevent it from being routed around.  I haven't located all the holes but it would be nice to ensure that any new routes are secure by default.

Reply:
We are considering disabling http by default, and adding a flag for teams to enable it with a documented caveat that auth has not been implemented and that this endpoint is experimental.

Reply:
@dr3s good point here, but I'm afraid I still don't have much idea how to configure auth on this controller. As a workaround, I have added a flag that disables auth on the Rest Controller's endpoints, which defaults to enabling auth that temporarily deals with the problem. Will open another issue for this.